processor/src/main/java/org/mapstruct/ap/internal/model/source/selector/XmlElementDeclSelector.java

```diff
@@ -112,4 +119,38 @@ else if ( nameMatches.size() > 0 ) {
             return methods;
         }
     }
+
+    private Pair getMatch(SourceMethod sourceMappingMethod, final String targetPropertyName) {
+        final TypeMirror startingMirror = sourceMappingMethod.getExecutable().getReturnType();
+        TypeMirror currentMirror = startingMirror;
+        TypeElement currentElement = sourceMappingMethod.getResultType().getTypeElement();
+
+        while (currentElement != null) {
+            for (Element enclosed : currentElement.getEnclosedElements()) {
+                if (enclosed.getKind().equals( ElementKind.FIELD ) && targetPropertyName != null
+                        && enclosed.getSimpleName().contentEquals( targetPropertyName )) {
+                    XmlElementRef xmlElementRef = enclosed.getAnnotation( XmlElementRef.class );
```

Comment:
Since prims are already used for XmlElementDecl in XmlElementDeclSelector I think I switch to prisms just for consistency.